benchmark/utils/general.hpp

```diff
@@ -0,0 +1,240 @@
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#ifndef GKO_BENCHMARK_UTILS_GENERAL_HPP_
+#define GKO_BENCHMARK_UTILS_GENERAL_HPP_
+
+
+#include <array>
+#include <functional>
+#include <map>
+#include <ostream>
+#include <random>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+
+#include <gflags/gflags.h>
+#include <rapidjson/document.h>
+#include <rapidjson/istreamwrapper.h>
+#include <rapidjson/ostreamwrapper.h>
+#include <rapidjson/prettywriter.h>
+
+
+// Global command-line arguments
+DEFINE_string(
+    executor, "reference",
+    "The executor used to run the benchmarks, one of: reference, omp, cuda");
+
+DEFINE_uint32(device_id, 0, "ID of the device where to run the code");
+
+DEFINE_bool(overwrite, false,
+            "If true, overwrites existing results with new ones");
+
+DEFINE_string(backup, "",
+              "If set, the value is used as a file path of a backup"
+              " file where results are written after each test");
+
+DEFINE_string(double_buffer, "",
+              "If --backup is set, this variable can be set"
+              " to a nonempty string to enable double"
+              " buffering of backup files, in case of a"
+              " crash when overwriting the backup");
+
+DEFINE_bool(detailed, true,
+            "If set, performs several runs to obtain more detailed results");
+
+DEFINE_uint32(seed, 42, "Seed used for the random number generator");
+
+DEFINE_uint32(warmup, 2, "Warm-up repetitions");
+
+DEFINE_uint32(repetitions, 10,
+              "Number of runs used to obtain an averaged result.");
+
+
+// Returns a random number engine
+std::ranlux24 &get_engine()
+{
+    static std::ranlux24 engine(FLAGS_seed);
+    return engine;
+};
+
+
+// helper for writing out rapidjson Values
+std::ostream &operator<<(std::ostream &os, const rapidjson::Value &value)
+{
+    rapidjson::OStreamWrapper jos(os);
+    rapidjson::PrettyWriter<rapidjson::OStreamWrapper> writer(jos);
+    value.Accept(writer);
+    return os;
+}
+
+
+// helper for setting rapidjson object members
+template <typename T, typename NameType, typename Allocator>
+void add_or_set_member(rapidjson::Value &object, NameType &&name, T &&value,
+                       Allocator &&allocator)
+{
+    if (object.HasMember(name)) {
+        object[name] = std::forward<T>(value);
+    } else {
+        auto n = rapidjson::Value(name, allocator);
+        object.AddMember(n, std::forward<T>(value), allocator);
+    }
+}
+
+
+// helper for splitting a comma-separated list into vector of strings
+std::vector<std::string> split(const std::string &s, char delimiter)
+{
+    std::istringstream iss(s);
+    std::vector<std::string> tokens;
+    for (std::string token; std::getline(iss, token, delimiter);
+         tokens.push_back(token))
+        ;
```

Comment:
I kind of prefer a `while`-loop over this because it is easier to understand what is going on: 
```C++
std::string token;
while(std::getline(iss, token, delimiter))
    tokens.push_back(token);
```
This `for`-loop practically does the same thing, but harder to understand